src/state/routing/types.ts

```diff
@@ -302,7 +338,99 @@ export class DutchOrderTrade extends IDutchOrderTrade<Currency, Currency, TradeT
   }
 }
 
-export type InterfaceTrade = ClassicTrade | DutchOrderTrade
+export class PreviewTrade {
+  public readonly fillType = TradeFillType.None
+  public readonly quoteMethod = QuoteMethod.QUICK_ROUTE
+  public readonly tradeType: TradeType
+  public readonly inputAmount: CurrencyAmount<Currency>
+  public readonly outputAmount: CurrencyAmount<Currency>
+  inputTax: Percent
+  outputTax: Percent
+
+  constructor({
+    inputAmount,
+    outputAmount,
+    tradeType,
+    inputTax,
+    outputTax,
+  }: {
+    inputAmount: CurrencyAmount<Currency>
+    outputAmount: CurrencyAmount<Currency>
+    tradeType: TradeType
+    inputTax: Percent
+    outputTax: Percent
+  }) {
+    this.inputAmount = inputAmount
+    this.outputAmount = outputAmount
+    this.tradeType = tradeType
+    this.inputTax = inputTax
+    this.outputTax = outputTax
+  }
+
+  public get totalTaxRate(): Percent {
+    return this.inputTax.add(this.outputTax)
+  }
+
+  public get postTaxOutputAmount() {
+    // Ideally we should calculate the final output amount by ammending the inputAmount based on the input tax and then applying the output tax,
+    // but this isn't currently possible because V2Trade reconstructs the total inputAmount based on the swap routes
+    // TODO(WEB-2761): Amend V2Trade objects in the v2-sdk to have a separate field for post-input tax routes
+    return this.outputAmount.multiply(new Fraction(ONE).subtract(this.totalTaxRate))
+  }
+
+  // copied from router-sdk Trade
```

Comment:
nit: can you link to it for reference? Also this comment isn't needed several times imo.

Reply:
bump.